editor/draw.js

```diff
@@ -111,17 +232,22 @@ svgedit.draw.Drawing = function(svgElem, opt_idPrefix) {
 	this.releasedNums = [];
 
 	/**
-	 * The z-ordered array of tuples containing layer names and <g> elements.
+	 * The z-ordered array of Layer objects. Each layer has a name
+	 * and group element.
 	 * The first layer is the one at the bottom of the rendering.
-	 * TODO: Turn this into an Array.<Layer>
-	 * @type {Array.<Array.<String, SVGGElement>>}
+	 * @type {Layer[]}
 	 */
 	this.all_layers = [];
```

Comment:
Layers are ordered but referenced externally by name, so we need both container types depending on which function is called. We can have one container and always build the other on each function call, but it's less code, and I think cleaner code, to keep both around.

I'll add a comment in the code about this.